### PR TITLE
[Notification] Cron trigger

### PR DESCRIPTION
**Pull request by qnkhuat**
Add cron trigger to the new notification system.
We still use Quartz as the scheduler backend, all tasks will be called with `notification/*send-notification*`, which then puts the task into a fixed thread pool to execute.

### CODE REVIEW

File: src/metabase/models/notification.clj [hunk 1]
```diff
@@ -62,17 +63,75 @@
    :id
    {:default nil}))
 
+(defn- delete-trigger-for-subscription!
+  [& args]
+  (classloader/require 'metabase.task.notification)
+  (apply (resolve 'metabase.task.notification/delete-trigger-for-subscription!) args))
```
Comment (piranha) [inline src/metabase/models/notification.clj:69]: This seems like a kind of `requiring-resolve`?

File: src/metabase/task/notification.clj [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  (triggers/key (format "metabase.task.notification.trigger.subscription.%d"
+                        subscription-id)))
+
+(defn- build-trigger
+  "Build a Quartz Trigger for `database` and `task-info` if a schedule exists."
+  ^CronTrigger [subscription-id cron-schedule]
+  (triggers/build
+   (triggers/with-description (format "Notification Subscription %d" subscription-id))
+   (triggers/with-identity (send-notification-trigger-key subscription-id))
+   (triggers/using-job-data {"subcription-id" subscription-id})
```
Comment (piranha) [inline src/metabase/task/notification.clj:40]: `subcription` :)

File: src/metabase/task/notification.clj [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        (notification/*send-notification!* (t2/select-one :model/Notification notification-id)))
+      (log/infof "Sent notification %d for subscription %d" notification-id subscription-id)
+      (catch Exception e
+        (log/errorf e "Failed to send notification %d for subscription %d" notification-id subscription-id)
+        (throw e)))))
+
+(jobs/defjob ^{:doc "Triggers that send a pulse to a list of channels at a specific time"}
+  SendNotification
+  [context]
+  (let [{:strs [subcription-id]} (qc/from-job-data context)]
```
Comment (piranha) [inline src/metabase/task/notification.clj:111]: `subcription`

File: test/metabase/models/notification_test.clj [hunk 4]
```diff
@@ -232,3 +236,71 @@
...
+   (task/existing-trigger @#'task.notification/send-notification-job-key
+                          (#'task.notification/send-notification-trigger-key subscription-id))))
+
+(defn- subscription->trigger-info
+  ([subscription-id cron-schedule]
+   (subscription->trigger-info subscription-id cron-schedule "UTC"))
+  ([subscription-id cron-schedule timezone]
+   {:key      (.getName (#'task.notification/send-notification-trigger-key subscription-id))
+    :schedule cron-schedule
+    :data     {"subcription-id" subscription-id}
```
Comment (piranha) [inline test/metabase/models/notification_test.clj:265]: and again :)

**Review by sansolovyov — APPROVED** [review]
just fix the subcriptions :)

**Pull request merged**